rasa/core/run.py

```diff
@@ -275,3 +279,19 @@ async def close_resources(app: Sanic, _: AbstractEventLoop) -> None:
     event_broker = current_agent.tracker_store.event_broker
     if event_broker:
         await event_broker.close()
+
+
+async def create_connections(
+    app: Sanic, _: AbstractEventLoop
+) -> Optional["ClientSession"]:
```

Comment:
What does this function do? It is named `create_connections` but it seems to be only returning a session

Reply:
We register this function as a [Sanic listener](https://sanic.dev/en/guide/basics/listeners.html#attaching-a-listener) that runs as part of the `after_server_start` event (see change in `rasa/core/run.py`). The wider context is that we need to establish this connection pool immediately after rasa server starts so that when a custom action runs, it doesn't increase latency of a bot response. Which was the issue @souvikg10 is validating.

Reply:
Can we improve the naming of this function then? Maybe `initialize_action_session`?

Reply:
I would maybe use `create_connection_pool` and make connection object for both `model_server` and `action_server`, since both http calls are happening a lot from rasa to action as well as to the model server.